tls/s2n_cipher_preferences.c

```diff
@@ -598,3 +609,22 @@ int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const cha
     return 0;
 }
 
+int s2n_connection_is_valid_for_cipher_preferences(struct s2n_connection *conn, const char *version)
+{
+    notnull_check(conn);
+    notnull_check(version);
+    notnull_check(conn->secure.cipher_suite);
+
+    const struct s2n_cipher_preferences *preferences;
+    GUARD(s2n_find_cipher_pref_from_version(version, &preferences));
+
+    for (int i = 0; i < preferences->count; ++i) {
+        if (0 == strcmp(preferences->suites[i]->name, conn->secure.cipher_suite->name) &&
+            /* make sure we dont use a tls version lower than that configured by the version */
+            conn->secure.cipher_suite->minimum_required_tls_version >= preferences->suites[i]->minimum_required_tls_version) {
```

Comment:
minimum_required_tls_version in cipher is used to verify if cipher is supposed to be available for this TLS version - it's not a strict enforcement on which protocol version connection needs to use. You need to check `preference->minimum_protocol_version` against `conn->actual_protocol_verison` to achieve what you want.